compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2514,7 +2514,10 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         def fullName: String = self.denot.fullName.toString
 
         def pos: Option[Position] =
-          if self.exists then Some(self.sourcePos) else None
+          if self.exists && !self.lexicallyEnclosingClass.isOneOf(dotc.core.Flags.JavaDefined | dotc.core.Flags.Permanent) then
+            Some(self.sourcePos)
+          else
+            None
```

Comment:
It seems weird to me that we perform this change, which is located outside of the Scaladoc codebase, in the implementation of quotes.

Looking at the discussion, there might be issues in the implementation of quotes, but that should not be a blocker for this PR. I suggest leaving the implementation of the quotes as it is (maybe open separate issues if you found bugs, @nicolasstucki), and defining a utility method in scaladoc itself like the following:

~~~ scala
def existingPosition(symbol: Symbol): Option[Position] = if self.exists && self.span.exists then Some(self.sourcePos) else None
~~~